File: java/sui-backend/src/main/java/ru/smsoft/sui/suibackend/config/ObjectMapperConfig.java
package ru.smsoft.sui.suibackend.config;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsonorg.JsonOrgModule;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import ru.smsoft.sui.suibackend.jackson.BackendJacksonModule;

import javax.annotation.PostConstruct;

@Configuration
@RequiredArgsConstructor
public class ObjectMapperConfig {

    @NonNull
    private ObjectMapper objectMapper;

    @PostConstruct
    private void configureObjectMapper() {
        objectMapper.registerModule(new JsonOrgModule());
        objectMapper.registerModule(new BackendJacksonModule());
        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
        objectMapper.getFactory().configure(JsonGenerator.Feature.ESCAPE_NON_ASCII, true);
    }

}

File: java/sui-utils/kotlin/src/main/java/ru/sui/utils/kotlin/config/ObjectMapperConfig.kt
package ru.sui.utils.kotlin.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.KotlinModule
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.annotation.PostConstruct


@Configuration
class MissingObjectMapperConfig {

    @Bean
    @ConditionalOnMissingBean
    fun simpleObjectMapper() = ObjectMapper()

}

@Configuration
class KotlinObjectMapperConfig(private val objectMapper: ObjectMapper) {

    @PostConstruct
    fun postConstruct() {
        objectMapper.registerModule(KotlinModule.Builder().build())
    }

}
